lib/util/http/auth/jwt.go

```diff
@@ -0,0 +1,77 @@
+package auth
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+//------------------------------------------------------------------------------
+
+// JWTConfig holds the configuration parameters for an JWT exchange.
+type JWTConfig struct {
+	Enabled       bool          `json:"enabled" yaml:"enabled"`
+	Claims        jwt.MapClaims `json:"claims" yaml:"claims"`
+	SigningMethod string        `json:"signing_method" yaml:"signing_method"`
+	PrivateKey    string        `json:"private_key" yaml:"private_key"`
```

Comment:
It often helps to make the fact that this is a file path explicit in the field name, so something like `private_key_file`.